servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletHttpResponse.java

```diff
@@ -0,0 +1,339 @@
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.ResponseHeadersBuilder;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Servlet response.
+ */
+final class DefaultServletHttpResponse implements HttpServletResponse {
+    private final List<Cookie> cookies = new ArrayList<>();
+    private final DefaultServletOutputStream outputStream;
+    private final ResponseHeadersBuilder headersBuilder = ResponseHeaders.builder();
+    private final PrintWriter writer;
+    private final HttpResponseWriter responseWriter;
+
+    DefaultServletHttpResponse(DefaultServletContext servletContext, HttpResponseWriter responseWriter) {
+        requireNonNull(servletContext, "servletContext");
+        requireNonNull(responseWriter, "responseWriter");
+        this.responseWriter = responseWriter;
+        outputStream = new DefaultServletOutputStream(this);
+        writer = new ServletPrintWriter(this, outputStream);
+        headersBuilder.set(HttpHeaderNames.CONTENT_TYPE, MediaType.HTML_UTF_8.toString());
+        setCharacterEncoding(servletContext.getRequestCharacterEncoding());
+    }
+
+    /**
+     * Get response writer.
+     */
+    HttpResponseWriter getResponseWriter() {
+        return responseWriter;
+    }
+
+    /**
+     * Parse cookies.
+     */
+    ResponseHeadersBuilder parseCookie() {
+        return headersBuilder.setObject(
+                HttpHeaderNames.SET_COOKIE,
+                cookies.stream()
+                       .map(x -> {
+                           final String path = x.getPath() == null ? "/" : x.getPath();
+                           final CookieBuilder builder =
+                                   com.linecorp.armeria.common.Cookie.builder(x.getName(), x.getValue())
+                                                                     .path(path)
+                                                                     .httpOnly(
+                                                                             x.isHttpOnly())
+                                                                     .secure(x.getSecure());
+                           if (x.getMaxAge() != -1) {
+                               builder.maxAge(x.getMaxAge());
+                           }
+                           if (!isNullOrEmpty(x.getDomain())) {
+                               builder.domain(x.getDomain());
+                           }
+                           return builder.build().toSetCookieHeader();
+                       }).collect(Collectors.toList()));
+    }
+
+    @Override
+    public void addCookie(Cookie cookie) {
+        requireNonNull(cookie, "cookie");
+        cookies.add(cookie);
+    }
+
+    @Override
+    public boolean containsHeader(String name) {
+        requireNonNull(name, "name");
+        return headersBuilder.contains(name);
+    }
+
+    @Override
+    public String encodeURL(String url) {
+        requireNonNull(url, "url");
+        return url;
```

Comment:
As Java doc said:
```
     * Encodes the specified URL by including the session ID,
     * or, if encoding is not needed, returns the URL unchanged.
     * The implementation of this method includes the logic to
     * determine whether the session ID needs to be encoded in the URL.
```
Therefore I return the URL unchanged because we have not support servlet session yet.

Reply:
But we call this from `encodeRedirectURL()`.
Shouldn't we encode it?

Reply:
Yes I think so. Tomcat's `encodeUrl()` also just return `Url` append `&jsessionId=....` : https://github.com/apache/tomcat/blob/master/java/org/apache/catalina/connector/Response.java . If we doesn't support servlet session now, I think we just return `Url` unchanged.

Reply:
Thanks for checking. 😄